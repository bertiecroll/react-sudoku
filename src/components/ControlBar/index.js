export { ControlBar } from './ControlBar'
export { default } from './ControlBar'
