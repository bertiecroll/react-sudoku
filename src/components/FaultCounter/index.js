export { FaultCounter } from './FaultCounter'
export { default } from './FaultCounter'
