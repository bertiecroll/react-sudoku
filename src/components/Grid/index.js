export { Grid } from './Grid'
export { GridRow } from './GridRow'
