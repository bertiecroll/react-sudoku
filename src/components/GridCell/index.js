export { GridCell } from './GridCell'
export { EmptyGridCell } from './EmptyGridCell'
export { CompleteGridCell } from './CompleteGridCell'
export { default } from './GridCell'
