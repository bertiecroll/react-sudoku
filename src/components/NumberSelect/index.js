export { NumberSelect } from './NumberSelect'
export { default } from './NumberSelect'
