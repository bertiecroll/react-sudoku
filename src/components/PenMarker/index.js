export { PenMarker } from './PenMarker'
export { default } from './PenMarker'
