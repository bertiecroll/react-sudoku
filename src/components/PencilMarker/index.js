export { PencilMarker } from './PencilMarker'
export { default } from './PencilMarker'
