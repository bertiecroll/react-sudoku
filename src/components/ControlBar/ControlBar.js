import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import NumberSelect from 'components/NumberSelect'
import SwitchButton from 'components/SwitchButton'
import PencilMarker from 'components/PencilMarker'
import { PEN_MARKER, currentMarker } from 'redux/selectors/controlBar'

export const ControlBar = props => {
  return (
    <div>
      <SwitchButton />
      { props.currentMarker === PEN_MARKER ? <NumberSelect /> : <PencilMarker /> }
    </div>
  )
}

const mapStateToProps = state => ({ currentMarker: currentMarker(state) })

export default connect(mapStateToProps)(ControlBar)

ControlBar.propTypes = {
  currentMarker: PropTypes.string.isRequired,
}
