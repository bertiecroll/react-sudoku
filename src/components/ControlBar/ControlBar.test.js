import React from 'react'
import { shallow } from 'enzyme'
import { ControlBar } from 'components/ControlBar'
import { PEN_MARKER, PENCIL_MARKER } from 'redux/selectors/controlBar'

describe('ControlBar', () => {
  it('renders SwitchButton with PenMarker when props.currentMarker is PEN_MARKER', () => {
    const wrapper = shallow(<ControlBar currentMarker={PEN_MARKER} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders SwitchButton with PencilMarker when props.currentMarker is PENCIL_MARKER', () => {
    const wrapper = shallow(<ControlBar currentMarker={PENCIL_MARKER} />)
    expect(wrapper).toMatchSnapshot()
  })
})
