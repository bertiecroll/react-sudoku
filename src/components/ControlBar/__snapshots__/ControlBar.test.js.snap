// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlBar renders SwitchButton with PenMarker when props.currentMarker is PEN_MARKER 1`] = `
<div
  className="ControlBar"
>
  <Connect(SwitchButton) />
  <Connect(PenMarker) />
</div>
`;

exports[`ControlBar renders SwitchButton with PencilMarker when props.currentMarker is PENCIL_MARKER 1`] = `
<div
  className="ControlBar"
>
  <Connect(SwitchButton) />
  <Connect(PencilMarker) />
</div>
`;
